File: uva_10533.cpp
# include <vector>
# include <iostream>
#define MAX 1000000
using namespace std;
vector<bool> prime(MAX, true);

bool is_digit_prime(int n){
    int sum = 0;
    while (n){
        sum+=n%10;
        n/=10;
    }
    if (prime[sum])
        return true;
    else
        return false;
}
int main(void){
    vector<int> digit_primes(MAX);
    for (int i=2; i<MAX; ++i){
        if (prime[i])
            for (int j=2*i; j<MAX; j+=i)
                prime[j]=false;
        if (prime[i] && is_digit_prime(i))
            digit_primes[i]=digit_primes[i-1]+1;
        else
            digit_primes[i]=digit_primes[i-1];
    }
    int num, t1, t2;
    cin >> num;
    for (int i=0; i<num; ++i){
        cin>>t1>>t2;
        cout << digit_primes[t2]-digit_primes[t1-1] << endl;
    }
    return 0;
}

File: uva_10533.py
MAX=1000000
prime=[True for _ in range(MAX)]
digit_prime=[0 for _ in range(MAX)]
def is_digit_prime(n):
    SUM=0
    while n:
        SUM+=n%10
        n=int(n/10)
    if prime[SUM]:
        return True
    else:
        return False

for i in range(2, MAX):
    if prime[i]:
        j=2*i
        while j<MAX:
            prime[j]=False
            j+=i
        if is_digit_prime(i):
            digit_prime[i]=digit_prime[i-1]+1
    if not digit_prime[i]:
        digit_prime[i]=digit_prime[i-1]

num=int(input())
for i in range(num):
    data=input().split()
    print(digit_prime[int(data[1])]-digit_prime[int(data[0])-1])
